server/src/main/java/io/druid/curator/CuratorModule.java

```diff
@@ -54,20 +59,25 @@ public void configure(Binder binder)
         binder, "druid.zk.service",
         CuratorConfig.class
     );
+    JsonConfigProvider.bind(
+        binder, "druid.exhibitor.service",
+        ExhibitorConfig.class
+    );
   }
 
   @Provides
   @LazySingleton
-  public CuratorFramework makeCurator(CuratorConfig config, Lifecycle lifecycle) throws IOException
+  public CuratorFramework makeCurator(
+      CuratorConfig config, ExhibitorConfig exConfig, Lifecycle lifecycle
+  ) throws IOException
   {
     final CuratorFramework framework =
-        CuratorFrameworkFactory.builder()
-                               .connectString(config.getZkHosts())
-                               .sessionTimeoutMs(config.getZkSessionTimeoutMs())
-                               .retryPolicy(new BoundedExponentialBackoffRetry(1000, 45000, 30))
-                               .compressionProvider(new PotentiallyGzippedCompressionProvider(config.getEnableCompression()))
-                               .aclProvider(config.getEnableAcl() ? new SecuredACLProvider() : new DefaultACLProvider())
-                               .build();
+        newCuratorBuilder(config, exConfig)
+            .sessionTimeoutMs(config.getZkSessionTimeoutMs())
+            .retryPolicy(new BoundedExponentialBackoffRetry(1000, 45000, 30))
```

Comment:
Can you move these magic numbers to static constants at the top of the class?

Eventually they should be moved to their own config entries.